doc/string/new.rdoc

```diff
@@ -1,34 +1,41 @@
-Returns a new \String that is a copy of +string+.
+Returns a new \String object containing the given +string+.
 
-With no arguments, returns the empty string with the Encoding <tt>ASCII-8BIT</tt>:
+The +options+ are optional keyword options (see below).
 
-  s = String.new
-  s # => ""
-  s.encoding # => #<Encoding:ASCII-8BIT>
+With no argument given and keyword +encoding+ also not given,
+returns an empty string with the Encoding <tt>ASCII-8BIT</tt>:
 
-With optional argument +string+ and no keyword arguments,
-returns a copy of +string+ with the same encoding:
+  s = String.new # => ""
+  s.encoding     # => #<Encoding:ASCII-8BIT>
 
-  String.new('foo')               # => "foo"
-  String.new('тест')              # => "тест"
-  String.new('こんにちは')          # => "こんにちは"
+With argument +string+ given and keyword option +encoding+ not given,
+returns a new string with the same encoding as +string+:
+
+  s0 = 'foo'.encode(Encoding::UTF_16)
+  s1 = String.new(s0)
+  s1.encoding # => #<Encoding:UTF-16 (dummy)>
+  s0 = ''.encode(Encoding::UTF_16)
+  s1 = String.new(s0)
+  s1.encoding # => #<Encoding:UTF-16 (dummy)>
 
 (Unlike \String.new,
 a {string literal}[rdoc-ref:syntax/literals.rdoc@String+Literals] like <tt>''</tt> or a
 {here document literal}[rdoc-ref:syntax/literals.rdoc@Here+Document+Literals]
 always has {script encoding}[rdoc-ref:encodings.rdoc@Script+Encoding].)
 
-With optional keyword argument +encoding+, returns a copy of +string+
-with the specified encoding;
+With keyword option +encoding+ given,
+returns a string with the specified encoding;
 the +encoding+ may be an Encoding object, an encoding name,
 or an encoding name alias:
 
+  String.new(encoding: Encoding::US_ASCII).encoding        # => #<Encoding:US-ASCII>
+  String.new('', encoding: Encoding::US_ASCII).encoding    # => #<Encoding:US-ASCII>
```

Comment:
What's the purpose of this example? It doesn't seem to be showing anything new.

Reply:
I was trying to show that the default empty string is different from as passed empty string.  I have removed the example and changed the call-seq.

Reply:
How is the default empty string is different from a passed in empty string? It works the same when you specify an `encoding`.